src/cpp/src/llm_pipeline.cpp

```diff
@@ -339,11 +394,354 @@ std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
 }  // namespace genai
 }  // namespace ov
 
+namespace {
+
+ov::Core get_core() {
+    static ov::Core core;
+    return core;
+}
+
+std::shared_ptr<ov::Model> add_slices_to_kvcache_inputs(const std::shared_ptr<ov::Model>& model) {
+    const auto kvcache_name_pattern = "past_key_values";
+    std::vector<std::shared_ptr<ov::opset13::Parameter>> new_params;
+    for (auto param : model->get_parameters()) {
+        auto tensor_name = param->get_output_tensor(0).get_any_name();
+        if (tensor_name.find(kvcache_name_pattern) == std::string::npos) {
+            new_params.push_back(param);
+            continue;
+        }
+        auto shape = param->get_output_shape(0);
+        shape[2] += 1;
+
+        auto new_param = std::make_shared<ov::opset13::Parameter>(param->get_element_type(), shape);
+        new_param->set_friendly_name(tensor_name);
+        new_param->outputs().begin()->get_tensor().set_names(param->outputs().begin()->get_tensor().get_names());
+
+        auto slice_start = std::make_shared<ov::opset13::Constant>(
+            ov::element::Type_t::i32, ov::Shape{1}, std::vector<int32_t>{1}
+        );
+        auto slice_stop = std::make_shared<ov::opset13::Constant>(
+            ov::element::Type_t::i32, ov::Shape{1}, std::vector<int32_t>{static_cast<int32_t>(shape[2])}
+        );
+        auto slice_step = std::make_shared<ov::opset13::Constant>(
+            ov::element::Type_t::i32, ov::Shape{1}, std::vector<int32_t>{1}
+        );
+        auto slice_axes = std::make_shared<ov::opset13::Constant>(
+            ov::element::Type_t::i32, ov::Shape{1}, std::vector<int32_t>{2}
+        );
+        auto slice_node = std::make_shared<ov::opset13::Slice>(
+            new_param, slice_start->output(0), slice_stop->output(0), slice_step->output(0), slice_axes->output(0)
+        );
+        slice_node->set_friendly_name(tensor_name + "_Slice");
+        for (auto target_input : param->output(0).get_target_inputs()) {
+            target_input.replace_source_output(slice_node->output(0));
+        }
+        new_params.push_back(new_param);
+    }
+    return std::make_shared<ov::Model>(model->get_results(), ov::SinkVector{}, new_params);
+}
+
+void reshape_to_static(std::shared_ptr<ov::Model> model,
+                       const uint32_t input_size,
+                       const uint32_t kvcache_size) {
+    std::map<std::string, ov::PartialShape> new_shapes;
+    for (auto input : model->inputs()) {
```

Comment:
`beam_idx` isn't handled there, as it's not present in models for NPU

Reply:
does it mean batch is not supported by NPU? 
Currently, we use `beam_idx` to drop sequences from batch, which have been finished. 
Or you can support it with padding even for finished sequences?

Reply:
Only batch=1 is supported, there is a special check for that in code. In general, I believe we still need to handle `beam_idx` in reshape function to comply with openvino models (not ours hand-crafted) but just set the shape of `[1]`.